### PR TITLE
feature/issue 151/x list detail view

### LINKED ISSUE #151: Implement detail view for a X-list
Implement a rough draft of the X-list (Krysseliste). Should be able to do the following
* Add an entry with product, amount and user as a list entry
* Edit existing entry
* Remove existing entry
All these actions should only be executable if a list is not closed.

There is not an existing design to this detail view, but in the first round it could prioritize a series list entries where each entry equates to a given post request. Bank accounts should not be charged before closing a list for simplicity's sake

### PR DESCRIPTION
**Pull request by alexaor**
Closes #151 

ToDo progress:
* [X] Implement product order ModelForm usage
* [X] Remove bank account balance information from source field selection 
* [X] Negative order sizes
* [X] Krysseliste name when `name` field is None
* [X] Product dropdown including more information
* [x] Re-implement overview view as not extremely ugly makeshift tables
* [x] Re-implement detail view as not extremely ugly makeshift tables
* [x] Minor pagination styling
* [x] Consider combining `closed`, `open` and `all` views into a single one  
* [x] Consider filtering away anyone not considered `active` in the ProductOrder.source dropdown (should maybe be postponed until organization module is properly dealt with)

### CODE REVIEW

File: economy/static/economy/css/economy_common.css [hunk 1]
```diff
@@ -16,8 +16,38 @@
...
-.economy__leftbutton.active{
+.economy__leftbutton.active {
     font-size: 24px;
     font-weight: 600;
     text-transform: uppercase;
+}
+
+.main-table {
+    display: flex;
+    flex-flow: column;
```
Comment (holtetx) [inline economy/static/economy/css/economy_common.css]: flex direction maybe?

File: economy/static/economy/css/economy_common.css [hunk 2]
```diff
@@ -16,8 +16,38 @@
...
+}
+
+.table-card > * {
+    width: 100px;
+}
+
+.table-header {
+    width: 100%;
+    display: flex;
+    flex-flow: row;
```
Comment (holtetx) [inline economy/static/economy/css/economy_common.css]: flex direction maybe?

File: economy/static/economy/css/economy_common.css [hunk 3]
```diff
@@ -16,8 +16,38 @@
...
+    justify-content: space-between;
+    margin-bottom: 20px;
+}
+
+.table-card {
+    padding: 0 15px;
+    margin: 4px 0;
+    background-color: #F0F0F0;
+    display: flex;
+    flex-flow: row;
```
Comment (holtetx) [inline economy/static/economy/css/economy_common.css]: flex direction maybe?

**Review by holtetx — DISMISSED** [review]
shalom shalom, ta en titt?

File: economy/static/economy/css/economy_common.css [hunk 4]
```diff
@@ -16,8 +16,38 @@
...
 }
 
-.economy__leftbutton.active{
+.economy__leftbutton.active {
     font-size: 24px;
     font-weight: 600;
     text-transform: uppercase;
+}
+
+.main-table {
```
Comment (trmdy) [inline economy/static/economy/css/economy_common.css]: It's slightly weird to name a flex-container "table".

If this is to be a tabular format, we should probably use a table element or a grid.

Comment (trmdy) [reply economy/static/economy/css/economy_common.css]: It seems the flex views also do not align. See

2020-12-03-181048_916x154_scrot

File: economy/views.py [hunk 5]
```diff
@@ -193,3 +201,26 @@ def soci_sessions_closed(request):
...
+        return redirect(reverse(soci_session_detail, kwargs={"soci_session_id": product_order.session.id}))
+    else:
+        return HttpResponse(status.HTTP_405_METHOD_NOT_ALLOWED)
+
+
+def product_order_add(request, soci_session_id):
+    if request.method == "POST":
+        session = get_object_or_404(SociSession, pk=soci_session_id)
+        if session and not session.closed:
+            form_data = request.POST  # rewrite as model form so we can get validation?
```
Comment (trmdy) [inline economy/views.py]: This should be a form, yes.

**Review by trmdy — CHANGES_REQUESTED** [review]
The code looks good. Some notes:

* All the flex views should most likely be grids or tables
* The dropdowns which mention "source" should probably mention people instead.
* The source dropdowns should probably not contain more information than the name of the person?
* The product dropdown should also have more concise contents.
* You can put negative order sizes.
* The titles of the "Krysseliste" seems to be consistently "None"?

We can also nitpick a bit on the design together if you'd like.

**Review by trmdy — APPROVED** [review]

**Pull request merged**